pkg/controllers/networkinfo/networkinfo_controller.go

```diff
@@ -118,9 +118,21 @@ func (r *NetworkInfoReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 		var privateIPs []string
 		var vpcConnectivityProfilePath string
-		if vpc.IsPreCreatedVPC(nc) {
+		var nsxLBSPath string
+		isPreCreatedVPC := vpc.IsPreCreatedVPC(nc)
+		if isPreCreatedVPC {
 			privateIPs = createdVpc.PrivateIps
 			vpcConnectivityProfilePath = *createdVpc.VpcConnectivityProfile
+			// Retrieve NSX lbs path if Avi is not used with the pre-created VPC.
+			if createdVpc.LoadBalancerVpcEndpoint == nil || createdVpc.LoadBalancerVpcEndpoint.Enabled == nil ||
```

Comment:
For AVI, it needs AVI subnet path, could be updated in a following patch

Reply:
Avi path is read here https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/controllers/networkinfo/networkinfo_controller.go#L184, the implemention is getting the Subnets from NSX directly using the provided VPC from NSX rather than from the local cache. So it works for both auto-created and pre-created VPC.